src/main/scala/firrtl/util/BackendCompilationUtilities.scala

```diff
@@ -24,17 +37,25 @@ trait BackendCompilationUtilities {
     out.close()
   }
 
-  /** Create a temporary directory with the prefix name. Exists here because it doesn't in Java 6.
+  /** Create a test directory
+    *
+    * Will create outer directory called testName then inner directory based on
+    * the current time
     */
-  def createTempDirectory(prefix: String): File = {
-    val temp = File.createTempFile(prefix, "")
-    if (!temp.delete()) {
-      throw new IOException(s"Unable to delete temp file '$temp'")
-    }
-    if (!temp.mkdir()) {
-      throw new IOException(s"Unable to create temp directory '$temp'")
+  def createTestDirectory(testName: String): File = {
+    val outer = new File(TestDirectory, testName)
+    val testDir = {
+      val dir: Future[File] = Future {
+        var _dir = new File(outer, timeStamp)
+        while (!_dir.mkdirs()) {
+          Thread.sleep(1)
+          _dir = new File(outer, timeStamp)
+        }
+        _dir
+      }
+      Await.result(dir, 10 seconds)
     }
```

Comment:
Can't you use this (in JDK >= 1.7):

    public static Path createTempDirectory(Path dir,
                           String prefix,
                           FileAttribute<?>... attrs)
                                    throws IOException

    Creates a new directory in the specified directory, using the given prefix to generate its
    name. The resulting Path is associated with the same FileSystem as the given directory.
    The details as to how the name of the directory is constructed is implementation
    dependent and therefore not specified. Where possible the prefix is used to construct
    candidate names.

    As with the createTempFile methods, this method is only part of a temporary-file facility. A
    shutdown-hook, or the File.deleteOnExit() mechanism may be used to delete the directory
    automatically.
    
    The attrs parameter is optional file-attributes to set atomically when creating the directory.
    Each attribute is identified by its name. If more than one attribute of the same name is
    included in the array then all but the last occurrence is ignored.